crates/core/component/funding/src/component/liquidity_tournament/votes/mod.rs

```diff
@@ -2,43 +2,195 @@ use std::pin::Pin;
 
 use async_trait::async_trait;
 use cnidarium::{StateRead, StateWrite};
-use futures::{Stream, StreamExt as _};
+use futures::{Stream, StreamExt as _, TryStreamExt as _};
 use penumbra_sdk_asset::asset;
 use penumbra_sdk_keys::Address;
+use penumbra_sdk_sct::component::source::SourceContext as _;
+use penumbra_sdk_txhash::TransactionId;
 
 use crate::component::state_key;
 
-pub trait StateReadExt: StateRead {
-    fn vote_receipts(
+async fn get_u64(state: impl StateRead, key: &[u8]) -> u64 {
+    state
+        .nonverifiable_get_raw(key)
```

Comment:
Should use `StateReadProto` and `StateWriteProto` so everything is uniformly encoded using the overlay

Reply:
I always forget that the primitive types implement the right traits 😅